cmake/alpakaCommon.cmake

```diff
@@ -0,0 +1,906 @@
+                    ENV HIP_PATH
+                    PATH_SUFFIXES "lib" "lib64")
+                if(NOT HIP_RAND_INC)
+                    message(FATAL_ERROR "Could not find hipRAND include (also searched in: HIP_ROOT_DIR=${HIP_ROOT_DIR}).")
+                endif()
+                if(NOT HIP_RAND_LIBRARY)
+                    message(FATAL_ERROR "Could not find hipRAND library (also searched in: HIP_ROOT_DIR=${HIP_ROOT_DIR}).")
+                endif()
+                target_include_directories(alpaka INTERFACE ${HIP_RAND_INC})
+                target_link_libraries(alpaka INTERFACE ${HIP_RAND_LIBRARY})
+            endif() # nvcc
+
+            if(ALPAKA_HIP_PLATFORM MATCHES "hcc")
+
+                # random numbers library ( HIP(HCC) ) /rocrand
+                find_path(ROC_RAND_INC
+                    rocrand_kernel.h
+                    PATHS "${HIP_ROOT_DIR}/rocrand" "${HIP_ROOT_DIR}" "rocrand"
+                    PATHS "/opt/rocm/rocrand"
+                    ENV HIP_PATH
+                    PATH_SUFFIXES "include")
+                find_library(ROC_RAND_LIBRARY
+                    rocrand-d
+                    rocrand
+                    PATHS "${HIP_ROOT_DIR}/rocrand" "${HIP_ROOT_DIR}" "rocrand"
+                    PATHS "/opt/rocm/rocrand"
+                    ENV HIP_PATH
+                    PATH_SUFFIXES "lib" "lib64")
+
+                # random numbers library ( HIP(HCC) ) rocrand/hiprand
+                find_path(HIP_RAND_INC
+                    hiprand_kernel.h
+                    PATHS "${HIP_ROOT_DIR}/hiprand" "${HIP_ROOT_DIR}" "hiprand"
+                    PATHS "/opt/rocm/hiprand"
+                    ENV HIP_PATH
+                    PATH_SUFFIXES "include")
+                find_library(HIP_RAND_LIBRARY
+                    hiprand-d
+                    hiprand
+                    PATHS "${HIP_ROOT_DIR}/hiprand" "${HIP_ROOT_DIR}" "hiprand"
+                    PATHS "/opt/rocm/hiprand"
+                    ENV HIP_PATH
+                    PATH_SUFFIXES "lib" "lib64")
+                if(NOT HIP_RAND_INC OR NOT HIP_RAND_LIBRARY)
+                    message(FATAL_ERROR "Could not find hipRAND library")
+                endif()
+
+                target_include_directories(alpaka INTERFACE ${HIP_RAND_INC})
+                target_link_libraries(alpaka INTERFACE ${HIP_RAND_LIBRARY})
+
+                if(NOT ROC_RAND_INC OR NOT ROC_RAND_LIBRARY)
+                    message(FATAL_ERROR "Could not find rocRAND library")
+                endif()
+
+                target_include_directories(alpaka INTERFACE ${ROC_RAND_INC})
+                target_link_libraries(alpaka INTERFACE ${ROC_RAND_LIBRARY})
+            endif()
+
+
+            list(APPEND HIP_HIPCC_FLAGS "-D__HIPCC__")
+            list(APPEND HIP_HIPCC_FLAGS "-std=c++${ALPAKA_CXX_STANDARD}")
+
+            if((${CMAKE_BUILD_TYPE} STREQUAL "Debug") OR (${CMAKE_BUILD_TYPE} STREQUAL "RelWithDebInfo"))
```

Comment:
must be `if((CMAKE_BUILD_TYPE STREQUAL "Debug") OR (CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo"))`